aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -293,19 +293,36 @@ pub struct AggregatorData {
     destroyed_aggregators: BTreeSet<AggregatorID>,
     // All aggregator instances that exist in the current transaction.
     aggregators: BTreeMap<AggregatorID, Aggregator>,
+    // If aggregators are not enabled, then all aggregators will directly store the Data instead of deltas.
+    is_aggregator_enabled: bool,
```

Comment:
I think this can live in aggregator context? Then if it's set you can query an aggregator and ensure it is `Data`

Reply:
I initially had the flag in the aggregator context. Then I realized that every time someone calls the `get_aggregator` flag, they will first need to get the `is_aggregator_enabled` flag from the context and then pass it as input to `get_aggregator` method. I thought the code would be more bug-prone if we forget to do this. What do you think?

Reply:
Hmm, I don't have a strong preference here. Having it in context and not in data ensures it is the caller who decides to materialize, so we know we do not have to re-read for example